### PR TITLE
Mvt symbology expression improvements

### PR DESCRIPTION
**Pull request by mhugent**
This PR adds the following improvements for the conversion of MVT styles to QGIS symbology:
- Support % and to-number
- keys can also be expressions
- Support conversion from string to color also if color is a subpart of an expression
- Support fill-opacity also if fill-color is present
- Add unit tests

### CODE REVIEW

File: src/core/vectortile/qgsmapboxglstyleconverter.cpp [hunk 1]
```diff
@@ -281,6 +283,7 @@ bool QgsMapBoxGlStyleConverter::parseFillLayer( const QVariantMap &jsonLayer, Qg
         if ( ddProperties.isActive( QgsSymbolLayer::PropertyFillColor ) )
         {
           context.pushWarning( QObject::tr( "%1: Could not set opacity of layer, opacity already defined in fill color" ).arg( context.layerId() ) );
```
Comment (nyalldawson) [inline src/core/vectortile/qgsmapboxglstyleconverter.cpp:285]: These warnings can be removed now, right?

**Comment by nirvn** [comment]
I'd consider backporting to 3.22. Those are small implementation gaps that are worth applying onto to-become 3.22 LTR. Similar improvements in the mapbox gl converter were recently backported to 3.22 too.

File: src/core/vectortile/qgsmapboxglstyleconverter.cpp (hunk 1 above)
Comment (mhugent) [reply src/core/vectortile/qgsmapboxglstyleconverter.cpp]: Yes, they are obsolete now

File: src/core/vectortile/qgsmapboxglstyleconverter.cpp [hunk 2]
```diff
@@ -2423,8 +2426,19 @@ QString QgsMapBoxGlStyleConverter::parseStops( double base, const QVariantList &
 
   const QVariant z = stops.last().toList().value( 0 );
   const QVariant v = stops.last().toList().value( 1 );
-  caseString += QStringLiteral( "WHEN @vector_tile_zoom > %1 "
-                                "THEN %2 END" ).arg( z.toString() ).arg( v.toDouble() * multiplier );
+  QString vStr = v.toString();
+  if ( ( QMetaType::Type )v.type() == QMetaType::QVariantList )
+  {
+    vStr = parseExpression( v.toList(), context );
+    caseString += QStringLiteral( "WHEN @vector_tile_zoom > %1 "
```
Comment (nyalldawson) [inline src/core/vectortile/qgsmapboxglstyleconverter.cpp:2431]: Can you add a test which covers this change?

**Comment by nyalldawson** [comment]
Lgtm!

File: src/core/vectortile/qgsmapboxglstyleconverter.cpp (hunk 2 above)
Comment (mhugent) [reply src/core/vectortile/qgsmapboxglstyleconverter.cpp:2431]: Done

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**